lib/flutter_chips_selector.dart

```diff
@@ -8,67 +8,80 @@ import 'package:flutter/scheduler.dart';
-      required this.suggestionBuilder,
-      required this.findSuggestions,
-      required this.onChanged,
-      this.parseOnLeaving,
-      this.decoration,
-      this.style,
-      this.autofocus,
-      this.keyboardBrightness = Brightness.light,
-      this.textInputType = TextInputType.text,
-      this.textInputAction = TextInputAction.done,
-      this.underlineColor = Colors.transparent,
-      this.labelColor,
-      FocusNode? currentFocus,
-      FocusNode? nextFocus})
-      : this.current = currentFocus ?? FocusNode(),
-        this.next = nextFocus ?? FocusNode(),
+  ChipsSelector({
+    Key? key,
+    this.initialValue = const [],
+    required this.chipBuilder,
+    required this.suggestionBuilder,
+    required this.findSuggestions,
+    required this.onChanged,
+    this.parseOnLeaving,
+    this.decoration,
+    this.style,
+    this.autofocus,
+    this.keyboardBrightness = Brightness.light,
+    this.textInputType = TextInputType.text,
+    this.textInputAction = TextInputAction.done,
+    this.underlineColor = Colors.transparent,
+    this.labelColor,
+    FocusNode? currentFocus,
+    FocusNode? nextFocus,
+  })  : this.textFieldFocusNode = currentFocus ?? FocusNode(), // TODO need to dispose
+        this.nextFocusNode = nextFocus ?? FocusNode(), // TODO need to dispose
         super(key: key);
 
-  final ChipsBuilder chipBuilder;
-  final ChipsBuilder suggestionBuilder;
+  final ChipsBuilder<T> chipBuilder;
+  final ChipsBuilder<T> suggestionBuilder;
   final ChipsInputSuggestions findSuggestions;
+
+  // TODO we should replace this with set since we enforce uniqueness in selectSuggestion anyways
   final List<T> initialValue;
+
+  // TODO we should rename this to onChangeChips or something like that
   final ValueChanged<List<T>> onChanged;
-  final ParsedItems? parseOnLeaving;
+
+  final ParsedItems<T>? parseOnLeaving;
   final InputDecoration? decoration;
   final TextStyle? style;
   final Color underlineColor;
   final Color? labelColor;
   final TextInputType? textInputType;
   final TextInputAction? textInputAction;
-  final FocusNode current;
-  final FocusNode next;
+  final FocusNode textFieldFocusNode;
+  final FocusNode nextFocusNode;
   final bool? autofocus;
   final Brightness keyboardBrightness;
 
+  @Deprecated("Use textFieldFocusNode instead")
+  FocusNode get current => textFieldFocusNode;
+
+  @Deprecated("Use nextFocusNode instead")
+  FocusNode get next => nextFocusNode;
+
   @override
   State<StatefulWidget> createState() => ChipsSelectorState<T>();
 }
 
-class ChipsSelectorState<T> extends State<ChipsSelector<T?>> {
+class ChipsSelectorState<T> extends State<ChipsSelector<T>> {
   final TextEditingController _textController = TextEditingController();
   final ScrollController _overlayScrollController = ScrollController();
   final GlobalKey editKey = GlobalKey();
-  late OverlayEntry _overlayEntry;
+  @visibleForTesting
+  OverlayEntry? suggestionOverlayEntry;
```

Comment:
Shouldn't we dispose this when the state get's disposed? Also we could remove the entry before disposing it

Reply:
definitely, changed